chat/public/js/src/chat.js

```diff
@@ -361,9 +361,8 @@ function ChatXBlock(runtime, element, init_data) {
      * pause: delays execution with a timeout
      */
     var pause = function(timeout) {
-        return new Promise(function(resolve, reject) {
-            setTimeout(resolve, timeout);
-        });
+        var promise = $.Deferred();
+        setTimeout(promise.resolve, timeout);
```

Comment:
Ah nice catch, thanks! I tried changing that, the unit test still fails I'm getting closer to fixing that.